### PR TITLE
PJR-200: OPDS 2.0 Types + Works and Manifestations

### PR DESCRIPTION
**Pull request by kristojorg**
This PR updates our `opds` package to add updated OPDS 2 types. Here are the changes:

1. Remove outdated opds2 types. These were used for our library registry compatibility (which we don't need anymore) and were incomplete. 
2. Adds OPDS 2 `Publication` type, and a new `WebpubManifest` type which extends the `Publication` to add a `readingOrder`, `resources`, and `toc` collections (the WebpubManifest is just a Publication with these added collection roles)
3. Adds OPDS 2 `Link` type, which is used in a `Publication`, `Feed`, `WebpubManifest`.
4. Adds OPDS 2 `Feed` type, with its own metadata, facets, groups and navigations.
5. Creates the `EpubWebpubManifest` and `PdfWebpubManifest` types, which are `WebpubManifest`s that conform to the profile of each (the PDF one is defined by us/DRB whereas the Epub one is defined by Readium).
6. Creates a draft `Work` profile, which is how we will extend the OPDS Publication type to incorporate the concept of Works and Manifestations. In the future we will also define `fulfillments` or `actions` - this may or may not be part of the same profile. There is a README for the work profile which ultimately can be included in a PR to Readium - though it is not ready yet.
7. Renames the `opds` package to `@nypl/opds` so that it can be published to NPM and used in the web reader.
8. Adds a `CHANGELOG.md` file.
9. Does not add a build script. We will publish raw Typescript src files to npm and the consuming package will have to bundle them.
10. Adds a `tsconfig.json` file to the `axisnow` package as the lack of one was causing a type error to appear in the root `tsconfig.json`.

## TO DO

1. Publish `@nypl/opds` to NPM (once this PR is approved - or could also make a new release PR once it is merged)
2. Use in the Web Reader

### CODE REVIEW

File: apps/pjr/android/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Open eBooks Hybrid Android
```
[the remark this answers is not in the document]
Comment (samanthaandrews) [reply apps/pjr/android/README.md:1]: probably just need to merge in `main`

File: apps/oew/src/components/theme-ui/AppLayoutPage.tsx [hunk 2]
```diff
@@ -3,15 +3,15 @@ import React, { ReactNode } from "react";
 import Page from "components/chakra-ui/Page";
 import AppLayout from "components/theme-ui/AppLayout";
 import { SearchData } from "interfaces";
-import { Collection } from "opds/opds2";
 
 /* LayoutPage is a Page with Header and Footer from Layout, this should be used to wrap pages within the app with sitewide navigation. */
 
-const AppLayoutPage: React.FC<React.PropsWithChildren<{
-  searchData?: SearchData;
-  collection?: Collection;
```
Comment (kristojorg) [inline apps/oew/src/components/theme-ui/AppLayoutPage.tsx:12]: collection was unused, so I removed it

File: packages/models/Book/types.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { OPDS1 } from "@nypl/opds";
+import { OPDS1 } from "../../opds/dist";
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply packages/models/Book/types.ts]: No you're right this should be importing from `@nypl/opds`. Must have been an autocomplete error or something

File: apps/oew/src/test-utils/fixtures/opds2.ts
Comment (kristojorg) [inline apps/oew/src/test-utils/fixtures/opds2.ts:1]: This was no longer needed

**Comment by kristojorg** [comment]
Ok I had to rebase and remove all the old commit history. I think that epub-to-webpub commit history is probably just lost

File: packages/axisnow/tsconfig.json
Comment (kristojorg) [inline packages/axisnow/tsconfig.json:1]: Added this as it was causing an error in the root tsconfig

File: packages/epub-to-webpub/src/convert/metadata.ts [hunk 4]
```diff
@@ -21,21 +15,22 @@ import { WebpubManifest } from '../WebpubManifestTypes/WebpubManifest';
...
-  const conformsTo = getConformsTo();
 
   return {
     title,
     language: language.length > 1 ? language : language[0],
     identifier,
     ...(presentation && { presentation }),
-    conformsTo,
-    ...contributors,
+    conformsTo: OPDS2.OPDS_EPUB_PROFILE,
```
Comment (kristojorg) [inline packages/epub-to-webpub/src/convert/metadata.ts:32]: conforms to is a constant so I got rid of the extra function

File: packages/opds/opds2/profiles/work/index.ts
Comment (kristojorg) [inline packages/opds/opds2/profiles/work/index.ts:1]: This is where I am defining a profile that describes our manifestations and fulfillments extensions

File: packages/epub-to-webpub/src/WebpubManifestTypes/ConformsTo.ts
[the remark this answers is not in the document]
Comment (kristojorg) [reply packages/epub-to-webpub/src/WebpubManifestTypes/ConformsTo.ts:1]: Yep, this became part of the opds2 package. Each profile now exports a conforms to that is specific to that profile

**Review by samanthaandrews — APPROVED** [review]
LGTM

File: packages/opds/opds2/Link/NavigationLink.ts
[the remark this answers is not in the document]
Comment (kristojorg) [reply packages/opds/opds2/Link/NavigationLink.ts:1]: Good call, I removed this for now. We currently have a general `Link` and an `AcquisitionLink` type, and we can add more specific link types as necessary

**Pull request merged**